src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1392,8 +1484,19 @@ private void PackageList_UpdateButtonClicked(PackageItemListViewModel[] selected
         private void ExecuteRestartSearchCommand(object sender, ExecutedRoutedEventArgs e)
         {
             EmitRefreshEvent(GetTimeSinceLastRefreshAndRestart(), RefreshOperationSource.RestartSearchCommand, RefreshOperationStatus.Success);
-            SearchPackagesAndRefreshUpdateCount(useCacheForUpdates: false);
-            RefreshConsolidatablePackagesCount();
+            ResetTabLoadFlags();
+            try
+            {
+                SearchPackagesAndRefreshUpdateCount(useCacheForUpdates: false);
+                RefreshConsolidatablePackagesCount();
+            }
+            catch (OperationCanceledException)
+            {
+                //Invalidate cache.
+                Model.CachedUpdates = null;
+                ResetTabLoadFlags();
+                throw;
```

Comment:
Should we be rethrowing here? 
Wouldn't this crash VS?

Reply:
If so, then we were already crashing VS 😨. I believe everything in here is FileAndForget, so this couldn't really happen. Likely just a leftover experiment that did nothing and I needed to get rid of.